gpflow/models/training_mixins.py

```diff
@@ -72,6 +72,12 @@ def training_loss_closure(self, *, compile=True) -> Callable[[], tf.Tensor]:
             return tf.function(self.training_loss)
         return self.training_loss
 
+    def after_data_changed(self) -> None:
```

Comment:
(there's something a bit ugly about the method name, though I'm not sure what. maybe the lack of a verb in the name of a method that returns `None`? how about: )

```suggestion
    def update_after_data_change(self) -> None:
```

Reply:
> (there's something a bit ugly about the method name, though I'm not sure what. maybe the lack of a verb in the name of a method that returns `None`? how about: )

Dunno. That's much too verbose for my taste. I was going for something that looks like a listener pattern. Maybe `on_data_change` or `data_changed`?

Reply:
> I mildly prefer `on_data_change` to `after_data_changed`, but also happy for you to leave it as it is.

I've renamed the method.